### DISCUSSION TITLE
How to use the saleor checkout app without Saleor Cloud?

### DISCUSSION THREAD

**Discussion by vinitjogani** [saleor/saleor#12871] [Q&A]
Hey, I am new here and am just trying to clone the Saleor React Storefront project and get a basic store running locally to play around with. I was able to set up the backend with docker easily. The frontend storefront also works fine. However, I can't seem to figure out how to get the checkouts to work without Saleor Cloud. All the commands/configs in the repo documentation just refer to CLI commands that only seem to work with the cloud. 

Steps I have tried that made logical sense:
- From the Events & Webhooks tab in the dashboard, I generated an app token for the checkout app
- Created a file called .saleor-app-auth.json with the token generated from above and placed it in the apps/saleor-app-checkout directory

The checkout UI opens but the network requests all seem to fail due to some APL issue. I fixed it by manually adding the auth token in the code like this:
```

export const get = async (saleorApiUrl: string) => {
  const authData: AuthData = {
    token: "...",
    domain: "http://localhost:8000/graphql/",
    saleorApiUrl: "http://localhost:8000/graphql/",
    appId: "QXBwOjI=",
    jwks: ""
  };
```

That seemed to work just enough to list the payment providers but nothing else after that. Payment processing still fails with the dummy app. 

Also, from the plugins section of the app, I tried adding Stripe and connected it to the default sales channel and that does not show up in these payment providers either.

I also see a bunch of other interesting apps on the store, is the only way to use it through Saleor Cloud? Can I integrate those apps if I am running locally on my own instance?

Any documentation about how to do this would be very helpful. I can only find Saleor Cloud related docs and they are very confusing.

**Comment by Belouh** [comment]
Hi. I have the same question and the same problem too.

**Comment by justJay-dev** [comment]
I had the same issue, the code snippet  @vinitjogani included works, but the root cause for me was I didn't have a `.env.local` i was expecting the root .env to override and it wasn't for me. when i made a `.env.local` with the appropriate variables loaded, the file APL stuff worked. 

Hope it helps.

**Comment by EnriqueRBT** [comment]
Did you get any solution?? Thanks

**Comment by maektwain** [comment]
I dont see any point using the checkout app, it's going to be deprecated soon.  Better try the new version of storefront.
